# Capítulo 2: Time Series

### Problema 1

a) La función de verosimilitud vendría dada por:
$$\mathcal{L} = \pi_{0,1} \times P_{12} \times P_{21} \times P_{12} \times P_{21}$$
Reemplazando se obtiene:
$$\mathcal{L} = 0.5 \times 0.1 \times 0.3 \times 0.1 \times 0.3$$

In [1]:
L = 0.5 * 0.1 * 0.3 * 0.1 * 0.3;
L

0.00045

b) La función de verosimilitud vendría dada por:
$$\mathcal{L} = \pi_{0,1} \times P_{11} \times P_{11} \times P_{11} \times P_{11}$$
Reemplazando se obtiene:
$$\mathcal{L} = 0.5 \times 0.9^4$$

In [2]:
L = 0.5 * 0.9^4;
L

0.32805000000000006

c) La función de verosimilitud vendría dada por:
$$\mathcal{L} = \pi_{0,2} \times P_{22} \times P_{22} \times P_{22} \times P_{22}$$
Reemplazando se obtiene:
$$\mathcal{L} = 0.5 \times 0.7^4$$

In [3]:
L = 0.5 * 0.7^4;
L

0.12004999999999996

### Problema 2

Podemos tratar de averiguar cuál es la expresión que representa $E[y^2_{t + 1} | x_t]$. En primer lugar tenemos:
$$E[y^2_{t + 1} | x_t = e_1] = (\bar{y}'e_1)^2 P_{11} + (\bar{y}'e_2)^2 P_{12}$$
Luego tenemos:
$$E[y^2_{t + 1} | x_t = e_2] = (\bar{y}'e_1)^2 P_{21} + (\bar{y}'e_2)^2 P_{22}$$

Entonces tendríamos que:
$$E[y^2_{t + 1} | x_t] = P \left[\begin{array}{c} 1 \\
                                                25\end{array}\right]$$

De manera análoga tendríamos que:
$$E[y_{t + 1} | x_t] = P \left[\begin{array}{c} 1 \\
                                                5 \end{array}\right]$$

Entonces se tendría el sistema:
$$\left[\begin{array}{cc} 1.8 & 5.8 \\
                          3.4 & 15.4 \end{array}\right] = P \left[\begin{array}{cc} 1 & 1 \\
                                                                                    5 & 25\end{array}\right]$$
$$J = Ph$$

La solución es única si el rango de la matriz $h'$ es completo o sea 2.

In [4]:
using LinearAlgebra

In [5]:
h = [1 1; 5 25]; J = [1.8 5.8; 3.4 15.4];

In [6]:
rank(h')

2

Entonces la matrix $P$ es única. La solución viene dada por:

In [7]:
X = h' \ J';
P = X'

2×2 Adjoint{Float64,Array{Float64,2}}:
 0.8  0.2
 0.4  0.6

### Problema 3

a) Sabemos que:
$$v_i = E\left[\sum^\infty_{t = 0} \beta^t u(c_t) | x_0 = \bar{e}_i\right] = \sum^\infty_{t = 0} E[\beta^t u(c_t) | x_0 = \bar{e}_i]$$

Por lo tanto, nos interesa saber que expresión representa $E[\beta^t u(c_t) | x_0 = \bar{e}_i]$. Tenemos que:
$$E[\beta^t u(c_t) | x_0 = \bar{e}_i] = \beta^t u(\bar{c}'\bar{e}_1)P^{(t)}_{i1} + \beta^t u(\bar{c}'\bar{e}_2)P^{(t)}_{i2} + \dots +\beta^t u(\bar{c}'\bar{e}_n)P^{(t)}_{in}$$ 

Entonces:
$$E[\beta^t u(c_t) | x_0 = \bar{e}_i] = \beta^t u_1 P^{(t)}_{i1} + \beta^t u_2 P^{(t)}_{i2} + \dots + \beta^t u_n P^{(t)}_{in}$$
Por lo que:
$$E[\beta^t u(c_t) | x_0] = \beta^t P^t u$$ 

Con esto se tiene que:
$$v = \sum^\infty_{t = 0} E[\beta^t u(c_t) | x_0] = \sum^\infty_{t = 0} \beta^t P^t u = (I - \beta P)^{-1}u$$

En cuanto a $V$ tenemos que:
$$V = E[v] = \pi'_0 v = \sum^n_{i = 1} \pi_{0,i} v_i$$

b) Construyamos funciones para: utilidad, $v$ y $V$:

In [14]:
u(c, γ) = c^(1 - γ) / (1 - γ) 

u (generic function with 3 methods)

In [15]:
v(P, β, u) = (I - β * P)^(-1) * u

v (generic function with 2 methods)

In [22]:
V(π_0, v) = π_0' * v

V (generic function with 1 method)

Construyamos el vector de utilidades en función del vector de consumo $\bar{c} = \left[\begin{array}{c} 1 \\
                                                                                                        5\end{array}\right]$ y $\gamma = 2.5$

In [18]:
c = [1, 5]; γ = 2.5;
util = u.(c, γ)

2-element Array{Float64,1}:
 -0.6666666666666666
 -0.05962847939999439

Consideremos el primer proceso:

In [21]:
P = [1 0; 0 1]; π_0 = [0.5, 0.5];

In [23]:
v_val = v(P, 0.95, util);
V_val = V(π_0, v_val)

-7.262951460666605

Consideremos el segundo proceso:

In [24]:
P = [0.5 0.5; 0.5 0.5]; π_0 = [0.5, 0.5];

In [25]:
v_val = v(P, 0.95, util);
V_val = V(π_0, v_val)

-7.262951460666601

Ambos procesos arrojan valores identicos $V_1 = V_2 = -7.26$ por lo que el consumidor es indiferente. Ahora analizemos el caso de $\gamma = 4$.

In [27]:
util = u.(c, 4.0)

2-element Array{Float64,1}:
 -0.3333333333333333
 -0.0026666666666666666

Consideremos el primer proceso:

In [28]:
P = [1 0; 0 1]; π_0 = [0.5, 0.5];
v_val = v(P, 0.95, util);
V_val = V(π_0, v_val)

-3.359999999999997

Ahora el segundo proceso:

In [29]:
P = [0.5 0.5; 0.5 0.5]; π_0 = [0.5, 0.5];
v_val = v(P, 0.95, util);
V_val = V(π_0, v_val)

-3.359999999999996

Nuevamente ambos procesos arrojan valores idénticos $V_1 = V_2 = -3.36$ por lo que el consumidor es indiferente.

c) Para el primer proceso la función de verosimilitud sería:
$$Prob(data | M_1) = \pi_{0,1} \times P^9_{11} = 0.5 \times 1^9$$

In [30]:
L_1 = 0.5 * 1^9

0.5

Para el segundo proceso la función de verosimilitud sería:
$$Prob(data | M_2) = \pi_{0,1} \times P^9_{11} = 0.5 \times 0.5^9 = 0.5^{10}$$

In [31]:
L_2 = 0.5^(10)

0.0009765625

d) Como las probabilidades iniciales de ambos modelos son idénticas tenemos que: $Prob(M_1) = Prob(M_2)$; por lo que:
$$Prob(M_1 | data) = \frac{Prob(data | M_1)}{Prob(data | M_1) + Prob(data | M_2)} = \frac{0.5}{0.5 + 0.001}$$

In [32]:
p1_n = L_1 / (L_1 + L_2)

0.9980506822612085

Para el caso del segundo proceso tenemos:
$$Prob(M_2 | data) = \frac{Prob(data | M_2)}{Prob(data | M_1) + Prob(data | M_2)} = \frac{0.001}{0.5 + 0.001}$$

In [33]:
p2_n = L_2 / (L_1 + L_2)

0.001949317738791423

e) Para la segunda muestra de datos, tenemos que:
$$Prob(data | M_1) = \pi_{0,1} \times P^4_{12} \times P^3_{21} \times P^2_{22} = 0.5 \times 0^4 \times 0^3 \times 1^2$$ 

In [34]:
L_1 = 0.5 * 0^4 * 0^3 * 1^2

0.0

Para el segundo proceso tenemos:
$$Prob(data | M_2) = \pi_{0,1} \times P^4_{12} \times P^3_{21} \times P^2_{22} = 0.5^{10}$$

In [35]:
L_2 = 0.5^(10)

0.0009765625

Tendríamos entonces que:
$$Prob(M_1 | data) = \frac{0}{0 + 0.001}$$

In [37]:
p1_n = L_1 / (L_1 + L_2)

0.0

Para el caso del segundo modelo:
$$Prob(M_2 | data) = \frac{0.001}{0 + 0.001}$$

In [38]:
p2_n = L_2 / (L_1 + L_2)

1.0

### Problema 4